.github/workflows/ci.yml

```diff
@@ -21,3 +21,28 @@ jobs:
       with:
         source: 'src test'
         extensions: 'ts'
+
+  publish:
+    name: Publish to VSCode marktplace
+    runs-on: ubuntu-latest
+    if: contains(github.ref, 'master') # publish it only if the push comes from master.
```

Comment:
I think it'd be nice to isolate this to a different workflow with different triggers, even if it means ignoring the CI.

If we want to run the tests, maybe we should do that explicitly (and redundantly) in the separate workflow, it's not much code.
I'm not sure it makes any sense to block releases on clang-format.